python/modules/variable.py

```diff
@@ -0,0 +1,381 @@
+            <parameter key = <KEY> value = <VALUE> type = <TYPE>/>
+    <parameter>
+    """
+    def __init__(self, key, value, args = {}):
+        Variable.__init__(self, key, value, args)
+
+    def get_variable(self, env):
+        converted_dict = self.get_value(env)
+        variable = VariableStructure(self.key, converted_dict)
+        return variable
+
+    def get_value(self, env):
+        converted_dict = {}
+        raw_dict = self.value
+        for key, var in raw_dict.items():
+            converted_dict[key] = var.get_variable(env)
+        return converted_dict
+        
+    def get_converted_value(self, env):
+        converted_dict = {}
+        for key, variable in self.value.items():
+            converted_dict[key] = variable.get_value(env)
+        return converted_dict
+
+    def create_dict(self, env):
+        return self.value
+
+class VariableMP(Variable):
+    """
+    VariableMP might represent
+    a) whole map_plate, where
+        value =  map_plate structure
+    b) single map_plate element (i.e. well), where
+        value = map_plate name (containing given map_plate element)
+
+    a) 
+    Map_plate structure (dictionary of ordered dictionaries)
+    presents as following example:
+        map_plate = { 'A01' : OrderedDict([('id', 'A01'),
+                                        ('exp_part', '1'),
+                                        ('name', 'A01'),
+                                        <...>]),
+                     'B01' : OrderedDict([('id', 'B01'),
+                                        ('exp_part', '2'),
+                                        ('name', 'A01'),
+                                        <...>]),
+                    <...> },
+        where:
+        map_plate.keys()- experimental wells
+        mp_plate[well_id]- dicionary of all experimental settings
+                            for given well
+
+    VariableMP representing whole map_plate is created during 
+    TASK_READ_MAP_PLATE.
+    
+    Variable can be refered in  xml settings
+    as following:
+    <parameter key = <KEY> value = <MP_NAME> type = "ref">
+
+    b)
+    Parameters with type 'map_plate', referring to chosen map_plate element
+    are represented in  xml settings as following example:
+    <parameter key = <KEY> value/mp_name = <MP_NAME> type = "map_plate">
+        <parameter key = "well" value = "mp_key" type = "ref">
+        <parameter key = "param" value = "stimulation.1.1">
+    </parameter>
+    """
+    def __init__(self, key, value, args = {}):
+        Variable.__init__(self, key, value, args)
+        self.mp_dict = value
+        
+    def get_mp_dict(self):
+        return self.mp_dict
+
+    def get_value(self, env):
+        args_keys = (self.args).keys()
+        if "well" in args_keys:
+            well = self.args["well"].get_value(env)
+            try:
+                self.mp_dict = env[self.value].get_mp_dict()
+            except:
```

Comment:
- To catch any exception use the ```except Exception```, instead of ```except BaseException``` or just ```except```.
- Indicate the most specific exception type in ```except``` block.

For example:
```python
# Bad
try:
    mapping[key]
except Exception:
    ...

# Better
try:
    mapping[key]
except KeyError:
    ...
```